pkg/pdp/tasks/watcher_eth.go

```diff
@@ -275,25 +282,31 @@ func (mw *MessageWatcherEth) checkTransaction(ctx context.Context, txHash common
 // getReceiptWithRetry fetches a transaction receipt with exponential backoff retry
 func (mw *MessageWatcherEth) getReceiptWithRetry(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
 	return backoff.Retry(ctx, func() (*types.Receipt, error) {
-		r, err := mw.api.TransactionReceipt(ctx, txHash)
+		txCtx, cancel := context.WithTimeout(ctx, mw.ethAPITimeout)
+		defer cancel()
+
+		r, err := mw.api.TransactionReceipt(txCtx, txHash)
 		// Don't retry on NotFound errors
 		if errors.Is(err, ethereum.NotFound) {
 			return nil, backoff.Permanent(err)
 		}
 		return r, err
-	}, backoff.WithMaxTries(mw.maxEthAPIRetries), backoff.WithBackOff(backoff.NewExponentialBackOff()))
+	}, backoff.WithMaxTries(mw.maxEthAPIRetries), backoff.WithBackOff(backoff.NewConstantBackOff(time.Second)))
```

Comment:
nb: are we sure we want constant backoff here?

Reply:
Fairly cretin, yes. Moved to a constant backoff to make it easier to reason about: The method will retry a maximum of 10 times, waiting one second between each failed attempt. The Lotus API will have three seconds to reply before we timeout the request due to cancellation.